tests/test_ldd.py

```diff
@@ -0,0 +1,220 @@
+#!/usr/bin/env python
+
+# pylint: disable=missing-docstring,too-many-public-methods
+import pathlib
+import unittest
+import subprocess
+
+import lddwrap
+
+
+def equals(dep: lddwrap.Dependency, other: lddwrap.Dependency) -> bool:
+    # Do not compare mem_address as it can differ between machines.
+    return dep.soname == other.soname and \
+           dep.path == other.path and \
+           dep.found == other.found and \
+           dep.unused == other.unused
+
+
+class TestLdd(unittest.TestCase):
+    def test_parse_line(self):
+        lines = [
+            "libstdc++.so.6 => /usr/lib/x86_64-linux-gnu/libstdc++.so.6 (0x00007f9a19d8a000)",
+            "/lib64/ld-linux-x86-64.so.2 (0x00007f9a1a329000)", "libboost_program_options.so.1.62.0 => not found",
+            "linux-vdso.so.1 =>  (0x00007ffd7c7fd000)", "libopencv_stitching.so.3.3 => not found",
+            "libstdc++.so.6 => /usr/lib/x86_64-linux-gnu/libstdc++.so.6 (0x00007f4b78462000)", "libz.so.1 => not found",
+            "../build/debug/libextstr.so => not found", "/home/user/lib/liblmdb.so => not found"
+        ]
+
+        expected_deps = [
+            lddwrap.Dependency(
+                soname="libstdc++.so.6",
+                path=pathlib.Path("/usr/lib/x86_64-linux-gnu/libstdc++.so.6"),
+                found=True,
+                mem_address=""),
+            lddwrap.Dependency(
+                soname=None, path=pathlib.Path("/lib64/ld-linux-x86-64.so.2"), found=True, mem_address=""),
+            lddwrap.Dependency(soname="libboost_program_options.so.1.62.0", path=None, found=False, mem_address=None),
+            lddwrap.Dependency(soname="linux-vdso.so.1", path=None, found=True, mem_address=""),
+            lddwrap.Dependency(soname="libopencv_stitching.so.3.3", path=None, found=False, mem_address=None),
+            lddwrap.Dependency(
+                soname="libstdc++.so.6",
+                path=pathlib.Path("/usr/lib/x86_64-linux-gnu/libstdc++.so.6"),
+                found=True,
+                mem_address=""),
+            lddwrap.Dependency(soname="libz.so.1", path=None, found=False, mem_address=None),
+            lddwrap.Dependency(
+                soname=None, path=pathlib.Path("../build/debug/libextstr.so"), found=False, mem_address=None),
+            lddwrap.Dependency(
+                soname=None, path=pathlib.Path("/home/user/lib/liblmdb.so"), found=False, mem_address=None)
+        ]
+
+        for i, line in enumerate(lines):
+            dep = lddwrap.parse_line(line=line)
+
+            self.assertTrue(
+                equals(dep, expected_deps[i]), "Incorrect dependency read from the line {}, expected {}\ngot {}".format(
+                    line, expected_deps[i], dep))
+
+    def test_parse_wrong_line(self):
+        line = "some wrong data which does not make sense"
+
+        self.assertRaises(Exception, lddwrap.parse_line, line)
```

Comment:
could you explain in the description what happened and check for specific excepion type?